app/controllers/bookings_controller.rb

```diff
@@ -0,0 +1,60 @@
+class BookingsController < ApplicationController
+  before_action :load_tour, except: %i(index change_status)
+  before_action :load_booking, only: %i(change_status)
+  before_action :not_logged_in, except: %i(new)
+  def new
+    @booking = Booking.new
+  end
+
+  def create
+    @booking = current_user.bookings.new booking_params
+    @booking.tour_id = @tour.id
+    @booking.price = @booking.quantity * @tour.price
+    if @booking.save
+      flash[:success] = t ".booking_success"
+      redirect_to bookings_path
+    else
+      flash[:danger] = t ".booking_fail"
+      render :new
+    end
+  end
+
+  def index
+    @bookings = current_user.bookings.order_by_create
+      .paginate page: params[:page], per_page: Settings.app.user.per_page
+  end
+
+  def change_status
+    @booking.cancel? ? @booking.waiting! : @booking.cancel!
```

Comment:
Booking ko có trạng thái reject hay approve ah chú

Reply:
dạ có chị ạ, mà đây là user sửa trạng thái nên em chỉ cho sửa waiting hay cancel thôi ạ, còn khi mà admin reject hay approve thì user không thể thay đổi trạng thái được nữa ạ.

Reply:
Ok.
Cơ mak chú check như ri thì...
Nếu là cancel thì update thành waiting, cơ mak ko phait cancel thì update toàn bộ là cancel.
Cái ko phải ở đây có thể là approve, reject hoặc waiting..
Nếu như ri chưa ổn lắm.
Vì mặc dù chú có check trên client view việc hiển thị button update hay ko, nhưng tui có thể hack dữ liệu :p

Nên check if-elsif-else
Else là trường hợp reject hoặc approve nhé:p

Reply:
dạ chị, chị xem thử như vậy hợp lý chưa ạ.